R/caching.R

```diff
@@ -81,7 +81,25 @@ copy_local_repo_to_cachedir <- function(local_dir, repo, host, verbose = 0) {
     message(paste("Copying local dir", local_dir, "to cache dir", repo_dir))
   }
   # file.copy copies a directory inside an existing directory
-  fs::dir_copy(local_dir, repo_dir)
+  # we ignore the renv sub directories as it is large (so slow), has long
+  # path names (so causes problems on Windows) and is not needed
+  # in the cache
+  fs::dir_create(repo_dir)
+  fs::file_copy(fs::dir_ls(local_dir, type = "file", all = TRUE), repo_dir)
+  # we also do not want the .Rprofile file
+  if (fs::file_exists(fs::path_join(c(repo_dir, ".Rprofile")))) {
+    fs::file_delete(fs::path_join(c(repo_dir, ".Rprofile")))
+  }
+
+  lapply(fs::dir_ls(local_dir, type = "directory", all = TRUE), function(dir) {
+    directory_to_copy <- utils::tail(strsplit(dir, .Platform$file.sep)[[1]], 1)
+    if(directory_to_copy != "renv") {
+      fs::dir_copy(dir, repo_dir)
+    } else {
+      fs::dir_create(fs::path_join(c(repo_dir, "renv")))
```

Comment:
do we need to copy any files in the `renv` directory at all?

Reply:
good point - we did when .rprofile was copied but as that's not we probably don't now - will check and update accordingly